src/components/autocompletes/multipleAutocomplete/multipleAutocomplete.tsx

```diff
@@ -0,0 +1,301 @@
+import styles from './multipleAutocomplete.module.scss';
+import { default as FieldText } from '@/components/fieldText';
+import { useFloating } from '@floating-ui/react';
+import { ControllerStateAndHelpers, PropGetters } from 'downshift';
+import { isEqual } from '../utils';
+import { ClearIcon } from '@/components/icons';
+
+const cx = classNames.bind(styles);
+
+export interface MultipleAutocompleteProps<T> {
+  options: T[];
+  loading: boolean;
+  onStateChange: () => void;
+  value: T[];
+  placeholder: string;
+  error: string | boolean; // string | boolean
+  touched: boolean;
+  creatable: boolean;
+  editable: boolean;
+  onChange: (selectedItems: T | T[] | null, downshift: ControllerStateAndHelpers<T> | null) => void;
+  onFocus: () => void;
+  onBlur: () => void;
+  disabled: boolean;
+  mobileDisabled: boolean;
+  inputProps: ComponentProps<typeof FieldText>;
+  parseValueToString: (value: T | null) => string;
+  minLength: number | null;
+  maxLength: number | null;
+  async: boolean;
+  customClass: string;
+  createWithoutConfirmation: boolean;
+  getItemValidationErrorType?: (item: T) => string;
+  clearItemsError: () => void;
+  getAdditionalCreationCondition: (value: T) => boolean;
+  highlightUnStoredItem: boolean;
+  parseInputValueFn: ((value: string) => T[]) | null;
+  handleUnStoredItemCb:
+    | ((newSelectedItems: DownshiftStore<T>, prevSelectedItems: DownshiftStore<T>) => void)
+    | null;
+  dataAutomationId: string;
+  existingItemsMap: { [key: string | number]: boolean };
+  variant: ComponentProps<typeof AutocompleteMenu>['variant'];
+  optionVariant: ComponentProps<typeof AutocompleteMenu>['optionVariant'];
+  isClearAvailable?: boolean;
+  customizeNewSelectedValue: (value: T) => T;
+  renderCustomSelecetedItem?: (item: T) => ReactNode;
+}
+
+export const MultipleAutocomplete = <T,>(componentsProps: MultipleAutocompleteProps<T>) => {
+  const {
+    options = [],
+    loading = false,
+    onStateChange = () => {},
+    value = [],
+    placeholder = '',
+    error = '',
+    touched = false,
+    creatable = false,
+    editable = false,
+    onChange = () => {},
+    onFocus = () => {},
+    onBlur = () => {},
+    disabled = false,
+    mobileDisabled = false,
+    inputProps = {},
+    parseValueToString = ((v: T) => v || '') as MultipleAutocompleteProps<T>['parseValueToString'],
+    minLength = 1,
+    maxLength = null,
+    async = false,
+    customClass = '',
+    createWithoutConfirmation = false,
+    getItemValidationErrorType = null,
+    clearItemsError = () => {},
+    getAdditionalCreationCondition = (() =>
+      true) as MultipleAutocompleteProps<T>['getAdditionalCreationCondition'],
+    highlightUnStoredItem = false,
+    parseInputValueFn = null,
+    handleUnStoredItemCb = null,
+    dataAutomationId = '',
+    existingItemsMap = {},
+    variant = 'light',
+    customizeNewSelectedValue = (v) => v,
```

Comment:
```suggestion
    customizeNewSelectedValue = (value) => value,
```
Quite strange default function.

Reply:
It's essentially a full copy of an existing component from Service-UI, but migrated to TypeScript and a few new libraries.